controllers/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -183,17 +186,20 @@ func (r *WorkspaceRoutingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	routesInSync, clusterRoutes, err := r.syncRoutes(instance, routes)
-	if err != nil || !routesInSync {
-		reqLogger.Info("Routes not in sync")
-		return reconcile.Result{Requeue: true}, err
-	}
-
 	clusterRoutingObj := solvers.RoutingObjects{
 		Services:  clusterServices,
 		Ingresses: clusterIngresses,
-		Routes:    clusterRoutes,
 	}
+
+	if r.IsOpenShift {
+		routesInSync, clusterRoutes, err := r.syncRoutes(instance, routes)
```

Comment:
Do you see any sense in syncing Ingresses on OpenShift?
I know they work somehow but how much we expect routings create ingresses on OpenShift?

Reply:
My idea was to not really worry about what we get back and just sync all the objects we can, which would mean that e.g. a solver that only generates ingresses would still work on OpenShift (hopefully). Otherwise, all solvers have to support routes instead of ingresses on OpenShift.

It's not terribly important, though, so I'll make the change.